pkg/protocol/types.go

```diff
@@ -37,3 +37,13 @@ const (
 	MosnHeaderMethod         = "method"
 )
 
+func init() {
+	RegisterProtocolFactories = make(map[types.Protocol]types.Protocol)
+}
+
+var RegisterProtocolFactories map[types.Protocol]types.Protocol
+
+func RegisterProtocol(protocol types.Protocol) {
+	//other
+	RegisterProtocolFactories[protocol] = protocol
+}
```

Comment:
I think this register is useless， the protocol implement just needs to define a types.Protocol is ok